src/main/java/DocumentIndexer.java

```diff
@@ -106,23 +103,26 @@ private void linkUses(DefinitionMeta meta, String documentId) {
         }
     }
 
-    private class LSIFVisitor extends CtScanner {
-        // TODO add support for more language constructs:
-        // https://github.com/INRIA/spoon/blob/master/src/main/java/spoon/reflect/visitor/CtScanner.java
+    // TODO add support for more language constructs:
+    // https://github.com/INRIA/spoon/blob/master/src/main/java/spoon/reflect/visitor/CtScanner.java
 
+    private class DefinitionsVisitor extends CtScanner {
```

Comment:
I split `LSIFVisitor` into 2 pieces:

- `DefinitionsVisitor`
- `ReferencesVisitor`

Most of the diff in this part of the PR is caused by me moving all the reference nodes to `ReferencesVisitor` and definition nodes to `DefinitionsVisitor`.